pkg/controller/kibana/config/settings.go

```diff
@@ -73,12 +89,57 @@ func NewConfigSettings(client k8s.Client, kb kbv1.Kibana, versionSpecificCfg *se
 	return CanonicalConfig{cfg}, nil
 }
 
+// getExistingConfig retrieves the canonical config for a given Kibana, if one exists
+func getExistingConfig(client k8s.Client, kb kbv1.Kibana) *settings.CanonicalConfig {
+	var secret corev1.Secret
+	err := client.Get(types.NamespacedName{Name: SecretName(kb), Namespace: kb.Namespace}, &secret)
+	if err != nil && apierrors.IsNotFound(err) {
+		log.V(1).Info("Kibana config secret does not exist", "namespace", kb.Namespace, "kibana_name", kb.Name)
+		return nil
+	} else if err != nil {
+		log.Error(err, "Error retrieving kibana config secret", "namespace", kb.Namespace, "kibana_name", kb.Name)
+		return nil
```

Comment:
I would expect we return the error here. Otherwise it means a failing `Get` call here (network reason or whatever) could end up with a successful `Update` later, and unexpected settings.

Reply:
Hmmm that makes sense. I'll change it to return an error here and exit the reconcile loop